lib/internal/modules/cjs/loader.js

```diff
@@ -1180,13 +1181,18 @@ function wrapSafe(filename, content, cjsModuleInstance) {
       '__dirname',
     ], {
       filename,
+      cachedData: codeCache,
       importModuleDynamically(specifier, _, importAssertions) {
         const cascadedLoader = getCascadedLoader();
         return cascadedLoader.import(specifier, normalizeReferrerURL(filename),
                                      importAssertions);
       },
     });
 
+    if (codeCache && result.cachedDataRejected !== false) {
```

Comment:
I am not sure if it is entirely necessary to throw here given that different V8 flags could also cause the cached data to be rejected, but then V8 is designed to silently ignore the rejected data and compile from scratch when that happens...so a warning here might be enough. And it would be useful to leave a comment here about this being SEA-only (or do add `isSEA()` into the conditions here).